vault/resource_identity_group.go

```diff
@@ -183,17 +184,18 @@ func identityGroupCreate(d *schema.ResourceData, meta interface{}) error {
 
 	if resp == nil {
 		path := identityGroupNamePath(name)
-		groupMsg := "Unable to determine group id."
-
-		if group, err := client.Logical().Read(path); err == nil {
-			groupMsg = fmt.Sprintf("Group resource ID %q may be imported.", group.Data["id"])
+		resp, err := client.Logical().Read(path)
+		if err == nil {
+			err = errors.New("unknown")
+			if resp != nil {
+				err = fmt.Errorf(
+					"group already exists with path=%q, id=%q", path, resp.Data["id"])
+			}
 		}
-
-		return fmt.Errorf("Identity Group %q already exists. %s", name, groupMsg)
-	} else {
-		log.Printf("[DEBUG] Wrote IdentityGroup %q", resp.Data["name"])
+		return fmt.Errorf("failed to create identity group %q, reason=%w", name, err)
```

Comment:
nit: any reason we are using `reason=%w` as opposed to `err=%w` like we are doing in other places in the code?

Reply:
Good question, it's largely due to the fact that `reason` may not have been derived from an actual `error`. For example when both `resp` and  `err` are nil. I am  happy to go with the other pattern as well.

Reply:
Got it, that makes sense